### PR TITLE
HSEARCH-4642 Bump dependencies / HSEARCH-4764 Bump AWS SDK to 2.19.8

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HSEARCH-4642
https://hibernate.atlassian.net/browse/HSEARCH-4764

For HSEARCH-4642, folllows up on https://github.com/hibernate/hibernate-search/pull/3339, https://github.com/hibernate/hibernate-search/pull/3322, https://github.com/hibernate/hibernate-search/pull/3313, https://github.com/hibernate/hibernate-search/pull/3294, https://github.com/hibernate/hibernate-search/pull/3146, https://github.com/hibernate/hibernate-search/pull/3148, https://github.com/hibernate/hibernate-search/pull/3157, https://github.com/hibernate/hibernate-search/pull/3176, https://github.com/hibernate/hibernate-search/pull/3189, https://github.com/hibernate/hibernate-search/pull/3194, https://github.com/hibernate/hibernate-search/pull/3208, https://github.com/hibernate/hibernate-search/pull/3219, https://github.com/hibernate/hibernate-search/pull/3234, https://github.com/hibernate/hibernate-search/pull/3239, https://github.com/hibernate/hibernate-search/pull/3249, https://github.com/hibernate/hibernate-search/pull/3255, https://github.com/hibernate/hibernate-search/pull/3262, https://github.com/hibernate/hibernate-search/pull/3281

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -232,7 +232,7 @@
         <version.org.opensearch.latest>2.4.1</version.org.opensearch.latest>
 
         <version.com.google.code.gson>2.9.1</version.com.google.code.gson>
-        <version.software.amazon.awssdk>2.18.1</version.software.amazon.awssdk>
+        <version.software.amazon.awssdk>2.19.8</version.software.amazon.awssdk>
```
Comment (yrodiere) [inline pom.xml:235]: @marko-bekhta this is not just a build/test dependency, it's exposed in published artifacts.

Can you please create a separate issue on JIRA and update the issue key in the commit that performs this update?
Also please mention that new issue in this PR title/description :)

**Review by yrodiere — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (marko-bekhta) [reply pom.xml:235]: sir yes sir 😄

Comment (yrodiere) [reply pom.xml:235]: Thank you sir :bow:

**Review by yrodiere — APPROVED** [review]

**Comment by yrodiere** [comment]
The test failure looks unrelated, I think it's about the failures we're having with ORM 6.2. I'll merge and we'll see what happens on the main branch.

**Comment by yrodiere** [comment]
Welp, the failure doesn't seem transient after all: https://hibernate.atlassian.net/browse/HSEARCH-4773

I doubt it's related to this PR though... probably one of your earlier PRs related to outbox-polling, @marko-bekhta ? Anyway, I've assigned the issue to you, please have a look when you have some time :)

**Pull request merged**